sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/OpenAIServerSentEvents.java

```diff
@@ -95,7 +85,12 @@ private void handleCurrentLine(String currentLine, List<T> values) throws JsonPr
             dataValue = split[1].substring(1);
         }
 
-        T value = SERIALIZER.readValue(dataValue, type);
+        T value = SERIALIZER.deserializeFromBytes(dataValue.getBytes(StandardCharsets.UTF_8), TypeReference.createInstance(type));
```

Comment:
This is great, I was actually wondering how this would look like with the stream API